src/main/java/com/android/volley/toolbox/DiskBasedCache.java

```diff
@@ -83,7 +81,8 @@ public File get() {
      *     briefly exceed this size on disk when writing a new entry that pushes it over the limit
      *     until the ensuing pruning completes.
      */
-    public DiskBasedCache(FileSupplier rootDirectorySupplier, int maxCacheSizeInBytes) {
+    public DiskBasedCache(
+            DiskBasedCacheUtility.FileSupplier rootDirectorySupplier, int maxCacheSizeInBytes) {
```

Comment:
Actually, this is an unsafe API change. This is a public constructor (so Volley clients may have used this), so we can't change the signature. (Moreover, DiskBasedCacheUtility is package private, so external callers can't even access it).

Unfortunately I think we need to leave FileSupplier here and duplicate it for the Async layer.

Reply:
This should be fixed now that FileSupplier is duplicated in DiskBasedCache